include/umf/memory_provider_ops.h

```diff
@@ -16,6 +16,64 @@
 extern "C" {
 #endif
 
+///
+/// @brief This structure comprises optional function pointers used
+/// by corresponding umfMemoryProvider* calls. A memory provider implementation
+/// can keep them NULL.
+///
+typedef struct umf_memory_provider_ext_ops_t {
```

Comment:
I'm wondering if it wouldn't be better to put allocation_merge and allocation_split in a separate struct and enforce user to set both of them or neither of them. Allowing setting only one of them would be error-prone.

Reply:
Perhaps we can think of a smarter validation check (e.g. check that both are set or both are NULL), but I am not sure that moving each pair of corresponding functions to a separate struct is a good idea. We might come up with too many optional structs.

Reply:
Ok, I don't think having more structs is necessarily a problem, but it might be tricky to name them so I guess we can leave it as is.